src/dodal/devices/i11/mythen.py

```diff
@@ -0,0 +1,165 @@
+
+
+class Mythen3TriggerMode(StrictEnum):
+    INTERNAL = "Internal"
+    EXTERNAL = "External"
+
+
+class Mythen3ModeSetting(StrictEnum):
+    STANDARD = "standard"
+    FAST = "fast"
+    HIGHGAIN = "highgain"
+
+
+class Mythen3DetectorState(StrictEnum):
+    """Default set of states of an  Mythen3 driver."""
+
+    IDLE = "Idle"
+    ERROR = "Error"
+    WAITING = "Waiting"
+    FINISHED = "Finished"
+    TRASMITTING = "Transmitting"
+    RUNNING = "Running"
+    STOPPED = "Stopped"
+    INITIALIZING = "Initializing"
+    DISCONNECTED = "Disconnected"
+    ABORTED = "Aborted"
+
+
+class Mythen3Driver(NDArrayBaseIO):
+    acquire_time: A[SignalRW[float], PvSuffix.rbv("AcquireTime")]
+    acquire_period: A[SignalRW[float], PvSuffix.rbv("AcquirePeriod")]
+    num_images: A[SignalRW[int], PvSuffix.rbv("NumImages")]
+    image_mode: A[SignalRW[ADImageMode], PvSuffix.rbv("ImageMode")]
+
+    # Non-specific PV's but with mythen3 specific values
+    detector_state: A[SignalR[Mythen3DetectorState], PvSuffix("DetectorState_RBV")]
+    trigger_mode: A[SignalRW[Mythen3TriggerMode], PvSuffix.rbv("TriggerMode")]
+
+    # mythen3 specific PV's
+    mode_setting: A[SignalRW[Mythen3ModeSetting], PvSuffix.rbv("Setting")]
+    bit_depth: A[SignalRW[int], PvSuffix.rbv("BitDepth")]
+    beam_energy: A[SignalRW[int], PvSuffix.rbv("BeamEnergy")]
+    threshold: A[SignalRW[int], PvSuffix.rbv("Counter1Threshold")]
+    threshold2: A[SignalRW[int], PvSuffix.rbv("Counter2Threshold")]
+    threshold3: A[SignalRW[int], PvSuffix.rbv("Counter3Threshold")]
+    global_threshold: A[SignalRW[int], PvSuffix.rbv("ThresholdEnergy")]
+
+
+"""
+Re. _DEADTIMES
+
+see: https://doi.org/10.1107/S1600577525000438
+Journal of Synchrotron Radiation, Volume 32, Part 2, March 2025, Pages 365-377
+See Section 3.3, Table 2, Section 3.4 and Table 3
+
+Note: Maximum frame rate of MYTHEN III is a function of the number of counters and of
+the bit depth.
+
+These numbers neglect possible limitations arising from the DAQ computing and network
+system. Expect ~1 order of magnitude less
+
+Our mythen3 is currently set up to use 3 counters, and 32bit depth. The deadtimes are
+for 3 counters only. We can run faster with less counters/bit depth
+
+The maximum frame rate of the detector scales both with the number of counters being
+read out and with the bit depth, which can be configured to 24 (streamed out as 32 bit),
+16 or 8 bits see Table 3.
+
+"""
+_DEADTIMES = {
+    32: 1 / (30 * 1000),
+    24: 1 / (40 * 1000),
+    16: 1 / (60 * 1000),
+    8: 1 / (120 * 1000),
+}
+
+_BIT_DEPTH = 32
+
+
+class Mythen3Controller(ADBaseController):
+    """ADBaseController` for a Mythen3"""
+
+    def __init__(self, driver):
+        self.driver = driver
+        super().__init__(driver=self.driver)
+
+    def get_deadtime(self, exposure: float | None) -> float:
+        return _DEADTIMES[_BIT_DEPTH]
+
+    async def prepare(self, trigger_info: TriggerInfo) -> None:
```

Comment:
Should: I think this needs a `@AsyncStatus.wrap`

Reply:
Why not? My understanding was that everything that ended up as a bluesky verb needed to be wrapped in an `AsyncStatus`, is this not true @coretl?

Reply:
`def StandardDetector.prepare() -> AsyncStatus` is a verb called from a plan that calls `async def DetectorController.prepare`. I agree that this is confusing, so happy for suggestions on what `DetectorController.prepare` should be named

Reply:
@olliesilvester, I will spin off an issue, we can discuss further there.